pkg/install/deployment.go

```diff
@@ -68,6 +69,14 @@ func WithEnvFromSecretKey(varName, secret, key string) podTemplateOption {
 	}
 }
 
+func WithSecretData(secretPresent bool) podTemplateOption {
+	return func(c *podTemplateConfig) {
+		c.withSecretData = secretPresent
```

Comment:
Do we need `withSecretData` in addition to `withoutCredentialsVolume`? Seems like they're the same thing.

Reply:
I've revised this so there's less duplication.